### ISSUE TITLE
Feature Request - Force disable match on upload

### ISSUE THREAD

**Issue by ScottESanDiego** [thebigmunch/google-music-scripts#24]
Marking a file as "Incorrectly Matched" in the WebUI, and then re-uploading with `gms` matches again.

Ideally there should be a flag to `gms sync up` and `gms upload` that disables matching, or a way to detect songs flagged "Incorrectly Matched" so they can be uploaded again w/o a match (as is done with the Windows Google Music Uploader).

**Comment by thebigmunch** [comment]
Nope. It's not possible to completely disable Google's scan-and-match system without unwanted side-effects. So adding an option to disable it that doesn't actually function completely is suspect.

Google can match a song in 3 ways (as best we can tell): metadata-only, client ID, and audio sample.

So, google-music-scripts already has a flag to send an empty audio sample. Its intent, however, has nothing to do with disabling scan-and-match, though it can possibly circumvent some matches.

Matching based on client ID is something that's possible to spoof. gmusicapi does this unintentionally as  it doesn't generate client IDs correctly (i.e. the same as Google) for the vast majority of files and wouldn't be able to reliably do so for anything but FLAC using mutagen, the audio metadata library gmusicapi uses. This is partially because we didn't know the exact method of calculation and partially because mutagen doesn't provide the information necessary. I actually did suss out exactly how it's done for FLAC (simple) and MP3 (not so simple) and have my own audio metadata library for this and many other reasons. Also, Google can catch on to gmusicapi's client IDs and start matching on those anyway in some cases.

The 3rd method is the one that happens the least, yet is probably the big show stopper. A song can match based purely on metadata. If you spoof the metadata (i.e. set all fields to empty strings), you get no metadata for the uploaded song on Google Music. There's no way around it. Trust me, I tried : P

If there were a way to properly circumvent all of Google's matching completely, I'd already have it implemented. There have been issues opened for gmusicapi due to songs matching despite matching being disabled. I'm not going to implement a feature/option that doesn't do what it says properly. It's unfortunate, because I certainly wanted this feature in as well. It'd be on permanently in my config.

-------------------------

As for the other part, are you saying that the Google Music Manager will force upload a song that has been reported as an incorrect match on the Web? (I've never actually had to report an incorrect match, nor really use Google's Music Manager outside testing) If so, I would expect these scripts to do the same unless there's something else possible in the calls for that I haven't seen yet.

**Comment by ScottESanDiego** [comment]
Interesting...thanks for the insight into the "Match" behavior.

I haven't seen the GMM behavior myself, but that's what is supposed to happen (e.g., per the thread https://www.reddit.com/r/google/comments/30e4du/issue_with_google_play_music_censoring_songs/ ).

**Comment by thebigmunch** [comment]
Well, I took a look through the old proto files I have, and it seems like there is definitely some stuff that could allow for something to be done. Unfortunately, I'd need to actually capture the traffic from Music Manager as it's happening. Music Manager needs to be patched to be able to capture traffic. Long story short, I would probably need to be using an older version of Windows to use the patched versions I have. So, it's mostly an idea of if/when I want to set up a VM to see if that's a possibility. It still might not be since I'd be throwing an exe from an older version on top of a newer version install, but we'll see when the time comes.
